sql/database.go

```diff
@@ -292,16 +328,201 @@ func Open(uri string, opts ...Opt) (*sqliteDatabase, error) {
 	return db, nil
 }
 
+func ensureDBSchemaUpToDate(logger *zap.Logger, db *sqliteDatabase, config *conf) (*sqliteDatabase, error) {
+	before, after, err := config.schema.CheckDBVersion(logger, db)
+	switch {
+	case err != nil:
+		return db, fmt.Errorf("check db version: %w", err)
+	case before == after:
+		return db, nil
+	case before > after:
+		// TODO: this should be logged by the caller
+		logger.Error("database version is newer than expected - downgrade is not supported",
+			zap.Int("current version", before),
+			zap.Int("target version", after),
+		)
+		return db, fmt.Errorf("%w: %d > %d", ErrTooNew, before, after)
+	case !config.enableMigrations:
+		// TODO: this should be logged by the caller
+		logger.Error("database version is too old",
+			zap.Int("current version", before),
+			zap.Int("target version", after),
+		)
+		return db, fmt.Errorf("%w: %d < %d", ErrOldSchema, before, after)
+	case config.temp:
+		// Temporary database, do migrations without transactions
+		// and sync afterwards
+		return db, config.schema.MigrateTempDB(logger, db, before)
+	case config.vacuumState != 0 &&
+		before <= config.vacuumState &&
+		strings.HasPrefix(config.uri, "file:"):
+		logger.Info("running migrations",
+			zap.Int("current version", before),
+			zap.Int("target version", after),
+		)
+		return db.copyMigrateDB(config)
+	default:
+		// Do not produce extra "running migrations" log message for the
+		// temporary DB, as it was already logged
+		if !config.temp {
+			logger.Info("running migrations in-place",
+				zap.Int("current version", before),
+				zap.Int("target version", after),
+			)
+		} else {
+			logger.Info("applying migrations to temporary DB",
+				zap.Int("current version", before),
+				zap.Int("target version", after),
+			)
+		}
```

Comment:
This doesn't need to be checked again:

```suggestion
		logger.Info("running migrations in-place",
			zap.Int("current version", before),
			zap.Int("target version", after),
		)
```

Reply:
Fixed